jenkinsci/tfs-plugin
# Associated work items link back to the Jenkins build

**NOTE:** The changes made in this branch were done on top of those of the `pull_request_merged_details` branch, already awaiting code review.  Once the associated pull request has been merged, this branch will be rebased and this note will be removed.

This branch introduces the `TeamUpdateWorkItemPostBuildAction`, which manifests itself as the **Add link to associated work items in TFS/Team Services** _post-build action_, illustrated in the screenshot below:

![image](https://cloud.githubusercontent.com/assets/297515/17684873/0effbeec-632d-11e6-8238-28a2c6cb47db.png)

Manual testing
--------------

1. Going to **Jenkins** > **Manage Jenkins** > **Configure System** > **TFS/Team Services**, one can now see a _description_ under the **Credentials** field: 
![image](https://cloud.githubusercontent.com/assets/297515/17685130/370632a2-632f-11e6-9e10-a352bf3adcfc.png)
...indeed, I did have to grant the personal access token the `work_write` permission before the next part worked.
2. Configure the **Git** and the **param** jobs to use the new action.
3. Craft a `git.pullrequest.merged` event payload that includes a reference to a Team Services work item (I used 640571) and send the payload to the `/team-events/gitPullRequestMerged` endpoint, then send the payload as part of a build command to the `/team-build/build/param` endpoint.
    1. Build of the **Git** and **param** job were queued.
    2. At the end of the builds, the new action was invoked.  Work item 640571 now has two new hyperlinks in its _Links_ section that point back to the 2 Jenkins builds queued in the previous step:

![image](https://cloud.githubusercontent.com/assets/297515/17685044/a19655b2-632e-11e6-8887-5af29f00c10a.png)
 

Mission accomplished!
